### PR TITLE
Add support for verifying ssl certificates 

### PR DESCRIPTION
**Pull request by Lukey3332**
## Description
For Security reasons it may be desirable to reject invalid certificates. This adds configuration
options to configure this.

Fixes: #594

## Types of changes
- [ ] Replay fix (fixes a replay specific issue)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added or updated tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by Lukey3332** [comment]
Weird, the tests pass on my side.

**Comment by Lukey3332** [comment]
Ping...

**Comment by ikreymer** [comment]
Hi, apologies for delay! There appears to be an issue appveyor currently.

This looks good, though I think want to make the `ca_certs_dir` optional. Can you default it to empty string and also a typo in the docs.

Thanks!

File: docs/manual/configuring.rst [hunk 1]
```diff
@@ -553,3 +553,14 @@ To enable the previous behavior, add to config::
...
 
 The system may be revamped in the future and enabled by default, but for now, it is provided "as-is" for compatibility reasons.
+
+Verify SSL-Certificates
+-----------------------
+
+By default, SSL-Certificates of websites are not verified. To enable verification, add the following to the config::
+
+  certificates:
+    cert_req: 'CERT_REQUIRED'
```
Comment (ikreymer) [inline docs/manual/configuring.rst:563]:
```suggestion
    cert_reqs:  'CERT_REQUIRED'
```

typo

File: pywb/warcserver/http.py [hunk 2]
```diff
@@ -17,8 +17,10 @@ class PywbHttpAdapter(HTTPAdapter):
     until a better solution is found
     """
 
-    # todo: allow configuring this later?
-    cert_reqs = 'CERT_NONE'
+    def __init__(self, cert_reqs='CERT_NONE', ca_cert_dir='/etc/ssl/certs', **init_kwargs):
```
Comment (ikreymer) [inline pywb/warcserver/http.py:20]:
```suggestion
    def __init__(self, cert_reqs='CERT_NONE', ca_cert_dir=None, **init_kwargs):
```

File: pywb/warcserver/warcserver.py [hunk 3]
```diff
@@ -70,6 +73,15 @@ def __init__(self, config_file='./config.yaml', custom_config=None):
 
         self.rules_file = self.config.get('rules_file', '')
 
+        if 'certificates' in self.config:
+            certs_config = self.config['certificates']
+            DefaultAdapters.live_adapter = PywbHttpAdapter(max_retries=Retry(3),
+                                                           cert_reqs=certs_config.get('cert_reqs', 'CERT_NONE'),
+                                                           ca_cert_dir=certs_config.get('ca_cert_dir', '/etc/ssl/certs'))
```
Comment (ikreymer) [inline pywb/warcserver/warcserver.py:80]:
```suggestion
                                                           ca_cert_dir=certs_config.get('ca_cert_dir'))
```

Want to make this optional.

File: pywb/warcserver/warcserver.py [hunk 4]
```diff
@@ -70,6 +73,15 @@ def __init__(self, config_file='./config.yaml', custom_config=None):
...
+            DefaultAdapters.remote_adapter = PywbHttpAdapter(max_retries=Retry(3),
+                                                             cert_reqs=certs_config.get('cert_reqs', 'CERT_NONE'),
+                                                             ca_cert_dir=certs_config.get('ca_cert_dir', '/etc/ssl/certs'))
```
Comment (ikreymer) [inline pywb/warcserver/warcserver.py:83]:
```suggestion
                                                             ca_cert_dir=certs_config.get('ca_cert_dir'))
```

File: docs/manual/configuring.rst [hunk 5]
```diff
@@ -553,3 +553,14 @@ To enable the previous behavior, add to config::
...
+    ca_cert_dir: '/etc/ssl/certs'
+
+``ca_cert_dir`` should point to a directory containing the CA certificates that you trust. Most linux distributions provide CA certificates via a package called ``ca-certificates``.
```
Comment (ikreymer) [inline docs/manual/configuring.rst:566]:
```suggestion
``ca_cert_dir`` can optionally point to a directory containing the CA certificates that you trust. Most linux distributions provide CA certificates via a package called ``ca-certificates``.
If omitted, the default system CA used by Python is used.
```

**Comment by Lukey3332** [comment]
Thank you for your Review. I addressed your comments.

**Comment by ikreymer** [comment]
Thanks! Could you add a quick test case, perhaps in the main `tests` directory?

You can just add a custom config yaml and then call a live endpoint, maybe something like: 
`/live/mp_/https://expired.badssl.com/` which should trigger the check?

You can look at a small test file like: `https://github.com/webrecorder/pywb/blob/master/tests/test_root_coll.py`
as an example.

**Comment by Lukey3332** [comment]
I added a testcase.

**Pull request merged**